clippy_lints/src/ptr.rs

```diff
@@ -251,12 +251,22 @@ impl<'tcx> LateLintPass<'tcx> for Ptr {
 
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) {
         if let ExprKind::Binary(ref op, l, r) = expr.kind {
```

Comment:
nit: you can drop the `ref` here since you're editing this region

```suggestion
        if let ExprKind::Binary(op, l, r) = expr.kind {
```

Reply:
removed